markdown-parsing.js

```diff
@@ -84,53 +102,69 @@ const filters: {[key: string]: (HTMLElement) => string | HTMLElement} = {
   },
   A(el) {
     const text = el.textContent
-    if (matches(el, 'issue-link', 'user-mention', 'team-mention')) {
+    const href = el.getAttribute('href')
+
+    if (matches(el, 'user-mention', 'team-mention')) {
       return text
-    } else if (/^https?:/.test(text) && text === el.getAttribute('href')) {
+    } else if (matches(el, 'issue-link') && /^#\d+$/.test(text)) {
+      return text
+    } else if (/^https?:/.test(text) && text === href) {
       return text
     } else {
-      const href = el.getAttribute('href')
-
-      if (!href) throw new Error()
-
-      return `[${text}](${href})`
+      if (href) {
+        return `[${text}](${href})`
+      } else {
+        return text
+      }
     }
   },
   IMG(el) {
-    const alt = el.getAttribute('alt')
-
-    if (!alt) throw new Error()
+    const alt = el.getAttribute('alt') || ''
 
-    if (matches(el, 'emoji')) {
+    if (alt && matches(el, 'emoji')) {
       return alt
     } else {
       const src = el.getAttribute('src')
-
       if (!src) throw new Error()
 
-      return `![${alt}](${src})`
+      const widthAttr = el.hasAttribute('width') ? ` width="${escapeAttribute(el.getAttribute('width') || '')}"` : ''
+      const heightAttr = el.hasAttribute('height')
+        ? ` height="${escapeAttribute(el.getAttribute('height') || '')}"`
+        : ''
+
+      if (widthAttr || heightAttr) {
+        return `<img alt="${escapeAttribute(alt)}"${widthAttr}${heightAttr} src="${escapeAttribute(src)}">`
+      } else {
+        return `![${alt}](${src})`
+      }
     }
   },
   LI(el) {
     const list = el.parentNode
 
     if (!list) throw new Error()
 
+    let bullet = ''
     if (!nestedListExclusive(el)) {
       switch (list.nodeName) {
         case 'UL':
```

Comment:
In the case of quoting from the middle of a list, `list` would be `#document-fragment`. Perhaps `UL` can be the default?

![image](https://user-images.githubusercontent.com/1153134/45426442-42e6a300-b66a-11e8-9e31-e5a5ae3ccd72.png)